src/libraries/System.Linq/src/System/Linq/Range.SpeedOpt.cs

```diff
@@ -33,9 +39,33 @@ public List<int> ToList()
 
             private static void Fill(Span<int> destination, int value)
             {
-                for (int i = 0; i < destination.Length; i++, value++)
+                Debug.Assert(Vector<int>.Count * sizeof(int) * 8 <= 512);
+
+                ref int pos = ref MemoryMarshal.GetReference(destination);
+                ref int end = ref Unsafe.Add(ref pos, destination.Length);
+
+                if (Vector.IsHardwareAccelerated && destination.Length >= Vector<int>.Count)
+                {
+                    Vector<int> init = new((ReadOnlySpan<int>)new int[] { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15 });
+                    Vector<int> current = new Vector<int>(value) + init;
+                    Vector<int> increment = new(Vector<int>.Count);
```

Comment:
I know why target-typed new isn't being used for the second line, but it looks inconsistent to use it for the first and third but not the second.  Please make them all `new Vector<int>`.